### PR TITLE
AAP-14003: Restructured the AAP 2.3 release notes

### PR DESCRIPTION
**Pull request by sayjadha**
This PR is to make similar release notes restructuring changes (done in AAP 2.4) in AAP 2.3 release notes.

JIRA: https://issues.redhat.com/browse/AAP-14003

Changes: 
As per the release notes restructuring efforts, following changes are made:
- Restructured AAP 2.3 release notes based on AAP versions instead of release date.
- Removed release numbers of components like controller, hub, and so on.
- Retained only 2.3 specific updates, and removed updates for 2.2, 2.1, and all earlier versions.

### CODE REVIEW

File: release-notes/topics/aap-23.adoc [hunk 1]
```diff
@@ -17,7 +17,11 @@ Red Hat Ansible Automation Platform simplifies the development and operation of
 * Updated internal FieldAttribute classes to act as Python data descriptors to reduce code complexity and use of metaclasses.
 * Fixed an issue when ansible-runner was not properly removed from hybrid nodes when upgrading to Ansible Automation Platform 2.2.
 
-.Technology preview features
+== Technology preview features
+
+Some features in this release are currently classified as Technology Preview. Technology Preview features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. Please note that Red Hat does not recommend using technology preview features for production, and Red Hat SLAs do not support technology preview functions.
```
Comment (ianf77) [inline release-notes/topics/aap-23.adoc:22]: Note that Red Hat does not recommend...

IBM style guide says not to use Please.

**Review by ianf77 — COMMENTED** [review]
One comment.

**Review by ianf77 — APPROVED** [review]
LGTM

File: release-notes/topics/aap-23.adoc (hunk 1 above)
Comment (sayjadha) [reply release-notes/topics/aap-23.adoc]: Made this change, thank you.

**Pull request merged**